prow/cmd/deck/user.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"k8s.io/test-infra/prow/userdashboard"
+)
+
+type userAgent struct {
+	path string
+
+	sync.Mutex
+	data *userdashboard.UserData
+}
+
+func (ua *userAgent) getData() (*userdashboard.UserData, error) {
+	ua.Lock()
+	defer ua.Unlock()
+
+	var data userdashboard.UserData
```

Comment:
nit: move this closer to where it is used

Reply:
done